rails/webpacker
# CSS Modules HMR working again

Demo: https://youtu.be/lIkhnLXIH6I

# Changes
1. Enable CSS modules HMR if webpack-dev-server running and HMR is on. Enabled by using the style-loader rather than extract CSS
2. Add docs for developing packages
3. export `inliningCss` from JS package
4. Remove inject_client webpacker.yml setting as this can be set by the HMR value
5. Added setting of injectClient and injectHot per the HMR setting
6. Simplifications and fixes for HMR 
   1. HMR default to true for webpacker.yml
   1. inline, injectClient, and injectHot default to the value of hmr
   1.  hmr turned on for the webpack dev server if the hmr config is true
   1.  using command line option for --hot as the docs recommend that over
     the plugin.

# To Test
Tested on https://github.com/shakacode/react_on_rails_tutorial_with_ssr_and_hmr_fast_refresh/pull/12

1. Checkout branch `justin808/change-css-loader`
2. bundle && yarn
3. Run `foreman start -f Procfile.dev-hmr`
4. Open up http://localhost:3000/hello_world and keep it visible to see the hot reloading
5. Edit the string on [line 17 of HelloWorld.tsx](https://github.com/shakacode/react_on_rails_tutorial_with_ssr_and_hmr_fast_refresh/blob/master/app/javascript/bundles/HelloWorld/components/HelloWorld.tsx#L17) and see the string change in the browser change upon saving the file.
6. Edit the text color on [line 2 of HelloWorld.module.css](https://github.com/shakacode/react_on_rails_tutorial_with_ssr_and_hmr_fast_refresh/blob/master/app/javascript/bundles/HelloWorld/components/HelloWorld.module.css#L2) and see the browser change upon saving the file.
